trogdor/src/main/java/org/apache/kafka/trogdor/workload/RoundTripWorker.java

```diff
@@ -14,447 +14,54 @@
-        }
-    }
-
-    private class ToReceiveTracker {
-        private final TreeSet<Long> pending = new TreeSet<>();
-
-        private long totalReceived = 0;
-
-        synchronized void addPending(long messageIndex) {
-            pending.add(messageIndex);
-        }
-
-        synchronized boolean removePending(long messageIndex) {
-            if (pending.remove(messageIndex)) {
-                totalReceived++;
-                return true;
-            } else {
-                return false;
-            }
-        }
-
-        synchronized long totalReceived() {
-            return totalReceived;
-        }
-
-        void log() {
-            long numToReceive;
-            List<Long> list = new ArrayList<>(LOG_NUM_MESSAGES);
-            synchronized (this) {
-                numToReceive = pending.size();
-                for (Iterator<Long> iter = pending.iterator();
-                        iter.hasNext() && (list.size() < LOG_NUM_MESSAGES); ) {
-                    Long i = iter.next();
-                    list.add(i);
-                }
-            }
-            log.info("{}: consumer waiting for {} message(s), starting with: {}",
-                id, numToReceive, list.stream().map(Object::toString).collect(Collectors.joining(", ")));
-        }
-    }
-
-    class ConsumerRunnable implements Runnable {
-
-        ConsumerRunnable(HashSet<TopicPartition> partitions) {
-            Properties props = new Properties();
-            props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, spec.bootstrapServers());
-            props.put(ConsumerConfig.CLIENT_ID_CONFIG, "consumer." + id);
-            props.put(ConsumerConfig.GROUP_ID_CONFIG, "round-trip-consumer-group-" + id);
-            props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
-            props.put(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG, 105000);
-            props.put(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, 100000);
-            // user may over-write the defaults with common client config and consumer config
-            WorkerUtils.addConfigsToProperties(props, spec.commonClientConf(), spec.consumerConf());
-            consumer = new KafkaConsumer<>(props, new ByteArrayDeserializer(),
+    public void initializeConsumer(HashSet<TopicPartition> partitions) {
+        Properties props = new Properties();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, spec.bootstrapServers());
+        props.put(ConsumerConfig.CLIENT_ID_CONFIG, "consumer." + id);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.put(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG, 105000);
+        props.put(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, 100000);
+        // user may over-write the defaults with common client config and consumer config
+        WorkerUtils.addConfigsToProperties(props, spec.commonClientConf(), spec.consumerConf());
+
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, "round-trip-share-group-" + id);
```

Comment:
`round-trip-share-group`- => `round-trip-consumer-group-`

Reply:
Thanks, I have updated this. I have also pasted output for `round_trip_json` workload with the changes.